server/gitrest/src/routes/git/refs.ts

```diff
@@ -64,14 +100,26 @@ async function patchRef(
     owner: string,
     repo: string,
     refId: string,
-    patchParams: IPatchRefParams): Promise<IRef> {
+    patchParams: IPatchRefParamsExternal,
+    externalStorageManager: IExternalStorageManager,
+): Promise<IRef> {
     const repository = await repoManager.open(owner, repo);
     const ref = await git.Reference.create(
         repository,
         refId,
         git.Oid.fromString(patchParams.sha),
         patchParams.force ? 1 : 0,
         "");
+
+    if (patchParams.config.enabled) {
+        try {
+            await externalStorageManager.write(repo, refId, patchParams.sha, true);
+        } catch (e) {
+            winston.error(`External storage write failed while trying to update file ${repo} / ${refId}`);
```

Comment:
Nit: Do you want to combine them into a single line

Reply:
Decided to keep it how it is because I think the line length will be exceeded otherwise